thrust/iterator/transform_iterator.h

```diff
@@ -312,6 +312,7 @@ template <class AdaptableUnaryFunction, class Iterator, class Reference = use_de
       // convert to their value type before calling m_f. Note that this
       // disallows non-constant operations through m_f.
       typename thrust::iterator_value<Iterator>::type const& x = *this->base();
+      // The issue is that x needs to be the type specified in m_f, but that is unknown 
       return m_f(x);
```

Comment:
I tried 
```
return static_cast<typename super_t::reference>(m_f(x));
```
To this one, but I believe it also caused some compile time errors

Reply:
Looked at this again, and not sure if there is a way (without adding extra args) to inherit the type expected by the unary operator m_f, might have to be ignored from -Wconversion?